### PR TITLE
fix(progressBar): IsIndeterminate don't trigger SetProgressBarIndicatorWidth when widths didn't change

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): closes #
#12312 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

- Indeterminate state was not porperly working on android

## What is the new behavior?

- Indeterminate state behave as expected

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by Xiaoy312 — COMMENTED** [review]
you should add a test for this
you should be able to validate the behavior by checking the number of calls to LayoutUpdated or VisualStateGroups["CommoonStates"].CurrentStateChanged

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs [hunk 1]
```diff
@@ -94,7 +95,11 @@ protected override void OnApplyTemplate()
...
 		private void OnSizeChange()
 		{
-			SetProgressBarIndicatorWidth();
+			var measures = (m_layoutRoot?.ActualWidth, m_determinateProgressBarIndicator?.ActualWidth);
+			if (m_previousMeasuredWidths != measures)
+			{
+				SetProgressBarIndicatorWidth();
+			}
 			UpdateWidthBasedTemplateSettings();
 		}
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs:104]: Is this a workaround for a different issue?

The original code doesn't have these checks:

https://github.com/microsoft/microsoft-ui-xaml/blob/4cbade6959cda4fca42a299fb540adc2f99c1e8a/dev/ProgressBar/ProgressBar.cpp#L50-L54

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs [hunk 2]
```diff
@@ -59,6 +59,7 @@ public ProgressBarTemplateSettings TemplateSettings
 		private Rectangle m_determinateProgressBarIndicator;
 		private Rectangle m_indeterminateProgressBarIndicator;
 		private Rectangle m_indeterminateProgressBarIndicator2;
+		private (double, double)? m_previousMeasuredWidths;
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs:62]: Let's use `Size` instead (it avoids the use of generics).

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs:104]: I meant the root cause for the issue might be something else and this is a workaround. It's unclear why it works on other platforms and not Android.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs [hunk 3]
```diff
@@ -94,7 +95,11 @@ protected override void OnApplyTemplate()
 
 		private void OnSizeChange()
 		{
-			SetProgressBarIndicatorWidth();
+			var measures = (m_layoutRoot?.ActualWidth, m_determinateProgressBarIndicator?.ActualWidth);
+			if (m_previousMeasuredWidths != measures)
+			{
+				SetProgressBarIndicatorWidth();
+			}
```
Comment (Xiaoy312) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs:102]:
```suggestion
#if __ANDROID__ // Uno workaround for #12312: SetProgressBarIndicatorWidth raises LayoutUpdated, and they many loops to stabilize
			var measures = (m_layoutRoot?.ActualWidth, m_determinateProgressBarIndicator?.ActualWidth);
			if (m_previousMeasuredWidths != measures)
#endif
			{
				SetProgressBarIndicatorWidth();
			}
```

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs:104]: The suggestion from @Xiaoy312 (https://github.com/unoplatform/uno/pull/12397#discussion_r1204305898) matches my expectation. This feels more like a workaround.

Comment (Robert-Louis) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs]: change made

**Comment by jeromelaban** [comment]
Let's ignore this new test on macos as well. (https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=2374423&_a=resultSummary&resultId=103658)

**Review by jeromelaban — APPROVED** [review]

**Review by Youssef1313 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Pull request merged**